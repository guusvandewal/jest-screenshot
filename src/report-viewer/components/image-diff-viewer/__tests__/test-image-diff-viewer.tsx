import * as React from "react";
import { shallow } from "enzyme";
import { ImageDiffViewer } from "..";

describe("ImageDiffViewer", () => {
    it("looks as expected", () => {
        expect(shallow(
            <ImageDiffViewer
                received="reports/something/received.png"
                snapshot="reports/something/snapshot.png"
                diff="reports/something/diff.png"
                width={100}
                height={100}
            />,
        )).toMatchSnapshot();
    });

    [0, 1, 2].forEach(index => {
        it(`when clicking on tab #${index}`, () => {
            const element = shallow(
                <ImageDiffViewer
                    received="reports/something/received.png"
                    snapshot="reports/something/snapshot.png"
                    diff="reports/something/diff.png"
                    width={100}
                    height={100}
                />,
            );
            const liElementsBeforeClick = element.find("li");
            for (let i = 0; i < liElementsBeforeClick.length; ++i) {
                expect(liElementsBeforeClick.at(i).hasClass("is-active")).toBe(i === 0);
            }
            element.find("li").at(index).find("a").simulate("click");
            expect(element).toMatchSnapshot();
            const liElementsAfterClick = element.find("li");
            for (let i = 0; i < liElementsAfterClick.length; ++i) {
                expect(liElementsAfterClick.at(i).hasClass("is-active")).toBe(i === index);
            }
        });
    });
});
