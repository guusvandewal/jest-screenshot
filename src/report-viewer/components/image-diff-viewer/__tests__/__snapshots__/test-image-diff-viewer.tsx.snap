// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDiffViewer looks as expected 1`] = `
<article>
  <div
    className="tabs is-toggle is-center is-fullwidth"
  >
    <ul>
      <li
        className="is-active"
      >
        <a
          onClick={[Function]}
        >
          Slider
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Side-by-side
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Blend
        </a>
      </li>
    </ul>
  </div>
  <DiffSlider
    diff="reports/something/diff.png"
    height={100}
    received="reports/something/received.png"
    snapshot="reports/something/snapshot.png"
    width={100}
  />
</article>
`;

exports[`ImageDiffViewer when clicking on tab #0 1`] = `
<article>
  <div
    className="tabs is-toggle is-center is-fullwidth"
  >
    <ul>
      <li
        className="is-active"
      >
        <a
          onClick={[Function]}
        >
          Slider
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Side-by-side
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Blend
        </a>
      </li>
    </ul>
  </div>
  <DiffSlider
    diff="reports/something/diff.png"
    height={100}
    received="reports/something/received.png"
    snapshot="reports/something/snapshot.png"
    width={100}
  />
</article>
`;

exports[`ImageDiffViewer when clicking on tab #1 1`] = `
<article>
  <div
    className="tabs is-toggle is-center is-fullwidth"
  >
    <ul>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Slider
        </a>
      </li>
      <li
        className="is-active"
      >
        <a
          onClick={[Function]}
        >
          Side-by-side
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Blend
        </a>
      </li>
    </ul>
  </div>
  <DiffSideBySide
    diff="reports/something/diff.png"
    height={100}
    received="reports/something/received.png"
    snapshot="reports/something/snapshot.png"
    width={100}
  />
</article>
`;

exports[`ImageDiffViewer when clicking on tab #2 1`] = `
<article>
  <div
    className="tabs is-toggle is-center is-fullwidth"
  >
    <ul>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Slider
        </a>
      </li>
      <li
        className=""
      >
        <a
          onClick={[Function]}
        >
          Side-by-side
        </a>
      </li>
      <li
        className="is-active"
      >
        <a
          onClick={[Function]}
        >
          Blend
        </a>
      </li>
    </ul>
  </div>
  <DiffBlend
    diff="reports/something/diff.png"
    height={100}
    received="reports/something/received.png"
    snapshot="reports/something/snapshot.png"
    width={100}
  />
</article>
`;
